Add redis/postgres in docker containers.  Refactor to use docker-compose directly

sunil-sadasivan opened the pull request:
After discussing with @mdbenjam, we decided to remove the simple abstractions around docker-compose and have developers to learn how to use it for orchestrating their local dependencies and containers. 

This PR does:
- Adds redis and postgress and VACOLS all under one docker-compose.yml in the root dir
- Refactors/Simplifies the local vacols initializer so that we remove the abstractions around docker-compose
- Moves the local vacols specific directory to `./local/vacols`

mdbenjam commented on lib/tasks/local/vacols.rake:
```diff
@@ -4,66 +4,26 @@ namespace :local do
     task setup: :environment do
-      Dir.chdir(Rails.root.join("vacols")) do
-        puts "Removing existing volumes"
-        `docker-compose down -v`
```
I think we should document the `docker-compose down -v` in the readme as a way to remove the volumes associated with the docker container in case you want to start over. Otherwise you can re-setup which I think might be useful.

sunil-sadasivan replied:
Yes good call, will do that!

mdbenjam commented on lib/tasks/local/vacols.rake:
```diff
@@ -4,66 +4,26 @@ namespace :local do
-        # Loop until setup is complete. At most 10 minutes
-        puts "Waiting for the database to be ready"
-        setup_complete = false
-        600.times do
```
Looking at this, the only thought I that this section will be useful for CI since we'll need a way to wait for the DB to be ready--but maybe that can be done in a separate rake task used for tests?

sunil-sadasivan replied:
Hmm, good call, I was going back and fourth whether I should clean this up or not.  I do like that this first ensures the db is ready before running, and in that case the abstraction around docker-compose I think makes sense.  I'll bring this back in.

sunil-sadasivan commented:
@mdbenjam updated a few more things. 

- Brought back the tear-down/bring-up steps in the vacols:setup command
- Updated the readme to include docker-compose down -v
- Adjusted the postgres compose definition to use the default volume location

mdbenjam commented on README.md:
````diff
@@ -93,25 +93,49 @@ cd /opt/oracle/instantclient_11_2
 ```
-rake local:vacols:setup
+export POSTGRES_USER=postgres
+export POSTGRES_PASSWORD=postgres
````
We should add

```
export POSTGRES_HOST=localhost
```

mdbenjam commented on README.md:
````diff
@@ -93,25 +93,50 @@ cd /opt/oracle/instantclient_11_2
+## Setup your Database Schema
 ```
-rake local:vacols:logs
+rake db:setup
````
Maybe we should include a [RAILS_ENV=<local|test|development>]? Though I think everyone already knows how to do that, it might be useful to alert folks to the local env.

mdbenjam commented on README.md:
```diff
@@ -93,25 +93,50 @@ cd /opt/oracle/instantclient_11_2
-To set up our local copy of VACOLS, run
+## Start up your docker based environment
+
+We use [docker](https://docs.docker.com/) and [docker-compose](https://docs.docker.com/compose/) to mock a production environment locally.  Prior knowledge of docker is not required, but slowly learning how docker works is encouraged.
```
For people setting this up after setting up pg and redis in the background they'll need to run:
```
brew services stop postgresql
brew services stop redis
```

This will stop them from launching on login.
